### DISCUSSION TITLE
Links to projects that use RE/flex

### DISCUSSION THREAD

**Discussion by genivia-inc** [Genivia/RE-flex#197] [Show and tell]
RE/flex has many use cases. It might be useful to point to projects that use RE/flex with a brief description how it is used. This may help others who are working on similar projects and to inspire new projects.

Let me start off by mentioning some of my open source projects that use RE/flex:
- [ugrep](ugrep.com) is a fast file pattern searcher, the [ugrep source code](https://github.com/Genivia/ugrep) is hosted on GitHub
- [lua-to-lisp](https://github.com/Robert-van-Engelen/lua-to-lisp) is a Lua 5.3 translator to Lisp (Common Lisp variants)
- [FuzzyMatcher](https://github.com/Genivia/FuzzyMatcher) approximate regex pattern matching (now also included in RE/flex)

Here are two interesting projects that I am not affiliated with, nor do I had any direct or indirect influence on:
- The [Tiger Compiler](https://assignments.lre.epita.fr) the Tiger Project as part of the [Epita](https://assignments.lre.epita.fr/tools/reflex.html) curriculum. It aims at the implementation of a Tiger compiler in C++.
- [Ox](https://www.complang.tuwien.ac.at/ublu/tools/doc/oxURM.pdf) is an attribute grammar compiling system that augments Lex and Yacc specifications with definitions of synthesized and inherited attributes written in a combination of Ox and C/C++ syntax. From these augmented specifications, Ox generates ordinary Lex and Yacc specifications that build and decorate attributed parse trees.

That's it for now. I know of many other small and large projects, but I don't want to take away your thunder if you want to reply to this post with a link and brief description of your project. Not all projects are open source, which is of course also fine to mention.

**Comment by ambroisie** [comment]
According to [the documentation](https://beancount.github.io/docs/beancount_v3_dependencies.html), [Beancount](https://beancount.github.io/) uses RE/Flex in it v3 rewrite for its UTF-8 support.
